### DISCUSSION TITLE
Add blocks to content.html

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#3476] [Ideas]
It seems weird that the `content.html` partial doesn't seem to have blocks, allowing to only edit selected sections of the theme through theme extension.

This requires me to copy the entirety of the content.html file, just to alter the icon used for the "Edit page" action.

This, of course, could be a wrong understanding of theme extensions on my end and if there is a way to change only parts of that file would I be happy to know about it, but other than that do I hope that some blocks for it could be added so that something like this could be doable:

```jinja2
{% block edit %}
  <a
    href="{{ page.edit_url }}"
    title="{{ lang.t('edit.link.title') }}"
    class="md-content__button md-icon"
  >
    {% include ".icons/octicons/pencil-24.svg" %}
  </a>
{% endblock %}
```

**Comment by squidfunk** [comment]
You can't add blocks inside partials, that's a limitation of Jinja.

**Reply by Andre601** [reply]
That's sad to hear...
Guess we can only hope jinja adds this in a future version...

**Reply by squidfunk** [reply]
I don't think this will be added, as partials may be included multiple times, even recursively, which is contrary to the singular nature of blocks.
